### PR TITLE
Code cleanup and support for table hints

### PR DESCRIPTION
**Pull request by Sicos1977**
I did some code cleanup and added support for MS SQL table hints.

If you don't like the code cleanup then just deny the PR and I'll make a new one with just the hints support

### CODE REVIEW

**Comment by ceastwood** [comment]
@ahmad-moussawi  I will try and help resolve the conflicts this evening

**Comment by ceastwood** [comment]
@Sicos1977 @Sicos2002  It looks like you may have a stale version of this repo-- I noticed because one of your commits just reverted changes @ahmad-moussawi had made to a unit test.

**Comment by Sicos1977** [comment]
I know I'll pull the changes upstream and merge them first to resolve the conflicts

**Comment by Sicos1977** [comment]
Conflicts resolved

**Comment by ceastwood** [comment]
@Sicos1977 Looks like something happened with QueryFactory.Extensions.cs as the changes don't build-- see the Travis-CI output for more detail. I can take a look tomorrow

**Comment by ahmad-moussawi** [comment]
@Sicos1977 what a generous contribution!, I have only one comment regarding omitting the `{}` for single if/else/for/etc... statements, keeping the braces will make the code more easier to read and less error prone in case of future updates

**Comment by Sicos1977** [comment]
Sorry about that. I use ReSharper and it was set to remove the braces. I'll try to avoid it.

**Comment by Sicos1977** [comment]
I'll place back the braces

**Comment by Sicos1977** [comment]
Also another question. Is it possible for you to add comment to the abstract classes? It's easier for me to understand what the code does and how everything hangs together.

**Comment by ahmad-moussawi** [comment]
Thanks, I am looking if we can find a service like https://styleci.io/ (for php) or http://jshint.com/ (obviously for js) so we can setup a coding style easily. Regarding the Abstract Class I am using them just as a type reference for certain type of clauses, so currently it's just a polymorphism thing.

In the next major version of SqlKata they will be removed, you can take a look at the expressions branch to see more details

**Comment by Sicos1977** [comment]
The braces should be back now :-). If you are looking for a good tool to enforce coding styles then look to ReSharper. You can get a free license for it if you have an opensource project that is used by many people.

https://www.jetbrains.com/buy/opensource/

**Comment by Sicos1977** [comment]
You still want me to do some more cleanup and add comment to make it easier for other developers to understand what a method does do?

**Comment by ahmad-moussawi** [comment]
Yes I have already a jetbrains license, but I am running on macos, so no luck with Sharper at the moment.

Regarding the comments yes off course and really appreciated

**Comment by ceastwood** [comment]
@ahmad-moussawi  VS Code is pretty nice on mac (after a handful of plugins) also

@ahmad-moussawi  @Sicos1977 

Whichever the service, it will probably end up using Style Cop rule sets; I think I remember one that integrates with GitHub, but will have to see if I can find the site.

On that topic, my only input right now with regard the cleanup are these rules about ordering (especially since we are adding regions):

http://stylecop.soyuz5.com/Ordering%20Rules.html

**Comment by ceastwood** [comment]
@Sicos1977 

I believe the class in the file SqlKata.Execution/QueryFactory.Extensions.cshas accidentally been renamed. It should be "QueryFactoryExtensions" which is causing the build errors.

**Comment by ahmad-moussawi** [comment]
@Sicos1977  are you able to resolve the error so I can merge the PR

**Comment by tlaguz** [comment]
@Sicos1977 Do you plan to finish this? I'm going to write Oracle compiler and would like to follow new code style

**Comment by Sicos1977** [comment]
Yeah sorry about that. I forgot that this was on my TO DO list. I'll try to fix it this evening.

**Comment by Sicos1977** [comment]
I think all the conflicts are gone now

File: QueryBuilder/Clauses/LimitClause.cs [hunk 1]
```diff
@@ -31,33 +40,55 @@ public int Offset
...
+        ///     Clears the limit and offset
+        /// </summary>
+        /// <returns></returns>
         public LimitOffset Clear()
         {
             return ClearLimit().ClearOffset();
         }
+        #endregion
 
+        #region Clone
```
Comment (tlaguz) [inline QueryBuilder/Clauses/LimitClause.cs]: What is the point of adding these regions?

File: QueryBuilder/Query.Delete.cs [hunk 2]
```diff
@@ -8,5 +8,16 @@ public Query AsDelete()
             return this;
         }
 
+        //public Query AsDeleteRaw(string expression, params object[] bindings)
```
Comment (tlaguz) [inline QueryBuilder/Query.Delete.cs]: Is this intended?

File: SqlKata.Execution/PaginationIterator.cs [hunk 3]
```diff
@@ -19,12 +16,16 @@ public IEnumerator<Execution.PaginationResult<T>> GetEnumerator()
...
-
         }
 
         IEnumerator IEnumerable.GetEnumerator()
         {
             return GetEnumerator();
         }
+
+        #region Properties
+        public PaginationResult<T> FirstPage { get; internal set; }
```
Comment (tlaguz) [inline SqlKata.Execution/PaginationIterator.cs]: As @ceastwood suggested: http://stylecop.soyuz5.com/Ordering%20Rules.html

File: SqlKata.Execution/Helper.cs [hunk 4]
```diff
@@ -9,39 +8,31 @@ internal static XQuery CastToXQuery(Query query, string method = null)
...
+                $"The method ${method} can only be used with `XQuery` instances, consider using the `QueryFactory.Query()` to create executable queries, check https://sqlkata.com/docs/execution/setup#xquery-class for more info");
         }
 
         internal static QueryFactory CreateQueryFactory(XQuery xQuery)
         {
-            var factory = new QueryFactory(xQuery.Connection, xQuery.Compiler);
-
-            factory.Logger = xQuery.Logger;
-
+            var factory = new QueryFactory(xQuery.Connection, xQuery.Compiler) {Logger = xQuery.Logger};
```
Comment (tlaguz) [inline SqlKata.Execution/Helper.cs]: I would move object initializer to the new line

**Pull request closed without merging**